### PR TITLE
Moving discordConstants functionality to dotenv

### PR DESCRIPTION
**Pull request by ProducerMatt**
**DEPENDS ON #247 and #263**

The goal of this PR is to externalize the discordConstants functionality to the
dotenv, so that the admin can customize it to their server.

# TODO

make all calls to discord for Rob server roles dependent on dotenv instead.
i.e. `discord.utils.get(guild.roles, name="bot dev")`

# Checked

- [ ] gpt3module (outdated and should be removed?)
- [ ] wolfram (need API access)
- [x] chatgpt (need api access)
- [x] questions (question retrieval and editing were split, thanks @MatthewBaggins)
- [x] videosearch
- [x] AlignmentNewsletterSearch
- [x] Eliza
- [x] Random
- [x] Silly
- [x] StampyControls
- [x] duckduckgo
- [x] sentience
- [x] why

irrelevant (or low-priority at least) outside of Rob's server: `invitemanager`,
`reply`, `stampcollection`

# Var Changes

summary of changes in config.py, **see lines 82 onward**

"Pluralizing" refers to going from singular variables, `if x == old_var` to
iterables checked like `if x in new_var`)

new:
- `ROBMILES_DEFAULTS`: when defined, sets relevant values to defaults for
  Rob's server and ignores other external settings, i.e. `bot_dev_ids`.
  Convenience setting for Rob.
- `DATABASE_PATH` for DB of Rob's YouTube comment questions
- `FACTOID_DATABASE_PATH` for `modules/Factoids.py`
- `BOT_DEV_IDS`: ids of dev users, union with `BOT_VIP_IDS`
- `BOT_REBOOT`: provides a choice for how the bot reboots itself.
- `PAID_SERVICE_FOR_ALL`: when defined, the ChatGPT module limits responses to an approved set of users
- `PAID_SERVICE_ROLE_IDS`: if the above is defined, these are the approved users

changed:
- `bot_admin_role_id,bot_dev_role_id` (latter unused) -> `BOT_DEV_ROLES`
  (pluralized)
- `rob_id` -> `BOT_VIP_IDS` (pluralized). I've also assumed that "VIP" user ids
  should be included in `BOT_DEV_IDS`
- `stampy_control_channel_ids` -> `BOT_CONTROL_CHANNEL_IDS`
- `stampy_private_channel_id` -> `BOT_PRIVATE_CHANNEL_ID`
- `stampy_id` -> instead please call function
  `Utilities.get_instance().stampy_is_author(message)`, this gets stampy's id
  from the current internal state rather than a config file
- `openai_channel_ids` -> `PAID_SERVICE_CHANNEL_IDS`

basically moved from Constants to config:
- `can_invite_role_id`
- `member_role_id`

### CODE REVIEW

**Comment by jknowak** [comment]
This would have to take into account the test I introduced here https://github.com/StampyAI/stampy/pull/253 @ProducerMatt

**Comment by ProducerMatt** [comment]
@jknowak thank you for bringing this to my attention.

Since my fork means people will enable different functionality and thus not need the same channels, I'll need to generalize this to some kind of module-updateable "needed discord channels" check that can alert the user if one is missing.

**Review by Aprillion — APPROVED** [review]
